src/Synapse/Synapse/Commands/DataPlaneCommands/AccessControl/NewAzureSynapseRoleAssignment.cs

```diff
@@ -124,7 +131,8 @@ public override void ExecuteCmdlet()
 
             if (this.ShouldProcess(this.WorkspaceName, String.Format(Resources.CreatingSynapseRoleAssignment, this.WorkspaceName, this.RoleDefinitionId, this.ObjectId)))
             {
-                PSRoleAssignmentDetails roleAssignmentDetails = new PSRoleAssignmentDetails(SynapseAnalyticsClient.CreateRoleAssignment(this.RoleDefinitionId, this.ObjectId));
+                string RoleAssignmentId = this.RoleDefinitionId + "-" + this.ObjectId;
```

Comment:
That is not valid GUID format? I know old RBAC using this format RoleAssignmentId, but SRBAC using standard GUID format instead of "GUID-GUID". We can generate a random GUID id or using a parameter to receive the value user setting.